packages/firestore/src/platform_node/grpc_connection.ts

```diff
@@ -95,26 +83,19 @@ export class GrpcConnection implements Connection {
   private firestore: any;
 
   // We cache stubs for the most-recently-used token.
-  private cachedStub: CachedStub | null = null;
+  private cachedStub: GeneratedGrpcStub | null = null;
 
   constructor(protos: grpc.GrpcObject, private databaseInfo: DatabaseInfo) {
     this.firestore = protos['google']['firestore']['v1'];
   }
 
-  private sameToken(tokenA: Token | null, tokenB: Token | null): boolean {
-    const valueA = tokenA && tokenA.authHeaders['Authorization'];
-    const valueB = tokenB && tokenB.authHeaders['Authorization'];
-    return valueA === valueB;
-  }
-
-  private ensureActiveStub(token: Token | null): GeneratedGrpcStub {
-    if (!this.cachedStub || !this.sameToken(this.cachedStub.token, token)) {
+  private ensureActiveStub(): GeneratedGrpcStub {
+    if (!this.cachedStub) {
```

Comment:
What consequences will removing this `||` statement have? Does this mean that we'll no longer reconnect with fresh credentials when a new token comes in?

Reply:
I'm hoping it has no consequences! 😄 

We used to associate the auth token with the stub (which might correspond to a gRPC channel?) which would allow all our RPCs to pick up the token for free.  And so in that world we had to recreate the stub whenever the token changed.  Hence this logic.

But after your rework, the reason for re-creating the stub went away because we no longer associate the token with the stub and so then we were just recreating the stub.

As for "no longer reconnect", gRPC manages the actual connection to the backend under the covers and I think even recreating the stub probably didn't trigger an actual TCP-level reconnect, though I don't have an easy way to verify that.